packages/nextjs/src/client-boundary/ClerkProvider.tsx

```diff
@@ -0,0 +1,19 @@
+'use client';
+
+import { useRouter } from 'next/compat/router';
```

Comment:
More details from the exported hook here: https://github.com/vercel/next.js/blob/canary/packages/next/src/client/compat/router.ts

Reply:
Is this a stable export we can rely on being available?

Or are we reaching into private APIs here?

Reply:
I believe it is stable and reliable, based on PR: https://github.com/vercel/next.js/pull/42502.